scripts/tests/test_zkai_native_attention_mlp_rmsnorm_adjacent_layout_gate.py

```
@@ -0,0 +1,79 @@
+        self.assertEqual(payload["adjacent_canonical_delta_vs_frontier_typed_bytes"], 248)
+        self.assertEqual(payload["adjacent_worst_label_typed_bytes"], 42_724)
+        self.assertEqual(payload["adjacent_worst_label_delta_vs_frontier_typed_bytes"], 2_024)
+        self.assertEqual(payload["adjacent_label_span_typed_bytes"], 1_776)
+        self.assertEqual(len(payload["mutation_results"]), 12)
+        self.assertTrue(all(entry["rejected"] for entry in payload["mutation_results"]))
+
+    def test_payload_rejects_frontier_overclaim(self) -> None:
+        payload = gate.build_payload()
+        payload["decision"] = "GO_FRONTIER_PROMOTION"
+        payload["payload_commitment"] = gate.payload_commitment(payload)
+        with self.assertRaises(gate.AdjacentLayoutGateError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)